the-10-min/src/Picker/GestureHandler.tsx

```diff
@@ -33,6 +34,7 @@ const GestureHandler = ({ value, max, defaultValue }: GestureHandlerProps) => {
     velocity: velocity.y,
     state,
     snapPoints,
+    offset: new Value(-ITEM_HEIGHT * 34),
   });
   useCode(() => [set(value, add(translateY, ITEM_HEIGHT * 2))], []);
```

Comment:
And for picker values I use this trick, onValueChange is prop

```
  useCode(() => [set(value, add(translateY, ITEM_HEIGHT * 2))], []);
  
  useCode(() => {
    return call([translateY], ([currentValue]) => {
      const selectedIndex = Math.round(-currentValue / ITEM_HEIGHT);
      const newValue = values[selectedIndex]?.value;

      if (onValueChange && newValue) {
        onValueChange(newValue, selectedIndex);
      }
    });
  }, [translateY]);
```

Reply:
Also keep in mind that you can't call setState in onValueChange callback It doesn't work, I use refs.
  const currentValueRef = React.useRef(defaultValue);
  
 <GestureHandler
                values={values}
                max={values.length}
                value={translateY}
                onValueChange={(value: any) => {
                  currentValueRef.current = value;
                }}
                defaultValue={defaultValue}
              />